ui/pages/notifications-settings/notifications-settings-allow-notifications.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState, useMemo, useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { useI18nContext } from '../../hooks/useI18nContext';
+import {
+  useEnableNotifications,
+  useDisableNotifications,
+} from '../../hooks/metamask-notifications/useNotifications';
+import {
+  selectIsMetamaskNotificationsEnabled,
+  getIsUpdatingMetamaskNotifications,
+} from '../../selectors/metamask-notifications/metamask-notifications';
+import { useMetamaskNotificationsContext } from '../../contexts/metamask-notifications/metamask-notifications';
+import { Box, Text } from '../../components/component-library';
+import {
+  Display,
+  JustifyContent,
+  FlexDirection,
+  AlignItems,
+  TextVariant,
+  TextColor,
+} from '../../helpers/constants/design-system';
+import {
+  NotificationsSettingsBox,
+  NotificationsSettingsType,
+} from '../../components/multichain';
+
+export function NotificationsSettingsAllowNotifications({
+  loading,
+  setLoading,
+  disabled,
+}: {
+  loading: boolean;
+  setLoading: (loading: boolean) => void;
+  disabled: boolean;
+}) {
+  const t = useI18nContext();
+  const { listNotifications } = useMetamaskNotificationsContext();
+  const isMetamaskNotificationsEnabled = useSelector(
+    selectIsMetamaskNotificationsEnabled,
+  );
+  const [toggleValue, setToggleValue] = useState(
+    isMetamaskNotificationsEnabled,
+  );
+  const isUpdatingMetamaskNotifications = useSelector(
+    getIsUpdatingMetamaskNotifications,
+  );
+
+  const { enableNotifications, error: errorEnableNotifications } =
+    useEnableNotifications();
+  const { disableNotifications, error: errorDisableNotifications } =
+    useDisableNotifications();
+  const error = errorEnableNotifications || errorDisableNotifications;
+
+  useEffect(() => {
+    setLoading(isUpdatingMetamaskNotifications);
+  }, [isUpdatingMetamaskNotifications, setLoading]);
+
+  useEffect(() => {
+    setToggleValue(isMetamaskNotificationsEnabled);
+  }, [isMetamaskNotificationsEnabled]);
+
+  useEffect(() => {
+    if (isMetamaskNotificationsEnabled && !error) {
+      listNotifications();
+    }
+  }, [isMetamaskNotificationsEnabled, error, listNotifications]);
```

Comment:
Nit/Future - I hate effects, effects are evil lol.

Ideally I want us to have a nice easy hook and try to avoid effects as much as possible.

Maybe some of these effects are justified, but not fully convinced.

Reply:
if it works, it works. But I want to tidy this in a future PR.

Reply:
Future PR about loading states.